scratchpad/bench/matmul.py

```diff
@@ -20,159 +20,34 @@
 exatn = LasyModule('exatn')
 
 
-class Matmul(Backend):    
+class MatmulBench(Benchmark):
+    @classmethod
+    def __init__(self, backend:Backend):
+        self.backend = backend
```

Comment:
Is it used anywhere? looks like you provide the backend already as an argument, I think it's better to just use one method: either argument of funciton or constructor